### DISCUSSION TITLE
Slow performance in pose mode even with viewport in Solid Mode?

### DISCUSSION THREAD

**Discussion by Dereuz24** [bnpr/Malt#584] [Q&A]
Hi there, I've been working in relatively complex scenes and I was wondering if the performance of a rig is supposed to be affected even when the viewport is not on render mode? I thought this was the case because Malt's pop up window is always on even when the viewport is set to solid mode, but is this the expected behaviour?

The way around has been to animate on eevee and just switch to Malt for visualization/modeling/material adjustments

**Comment by pragma37** [comment]
It shouldn't affect performance, but that doesn't mean Malt is not doing anything even if it's not rendering.

- Check the Blender System Console (Toggle System Console). If there's some error printing all the time that may affect performance.
- Check the Malt process CPU usage, if it's too high there may be something wrong.
- Malt needs to track Blender ID updates, this should be pretty fast, but if there are a lot of IDs being updated it may affect performance.
  If the overhead is on the addon side, this should point at the cause: https://github.com/pragma37/Blender-Python-Profiler

**Reply by Dereuz24** [reply]
Thanks!

I ran your tool while in object mode and indeed it was Malt tracking  shader changes and things like that. The overhead was like 30% more compared to eevee but it's a large scene so Imagine that's why it's noticeable

I'm animating on EEVEE with everything simplified so this is not really an issue, just wanted to know how it works
